commands/root.go

```diff
@@ -90,3 +95,40 @@ func Execute() {
 		os.Exit(1)
 	}
 }
+
+func getTagsAndCommit() (tag, exacttag, commit string) {
+	var t, e, c string
+
+	info, ok := debug.ReadBuildInfo()
+
+	if !ok {
+		fmt.Println("could not get commit")
+	}
+
+	for _, kv := range info.Settings {
+		switch kv.Key {
+		case "vcs.revision":
+			c = kv.Value
+		}
+	}
+
+	cmd := exec.Command("git", "describe", "--tags", "--abbrev=0")
```

Comment:
I'm not sure we want to run this every time a `git-bug` command is executed and, since we're already using `go-git` using `exec` might be a bit dangerous.  Theoretically, `git-bug` is run inside a `git` repository and the executable should be available but perhaps not?

The bigger issue is that this would return tag information for the `git` repository where `git-bug` is run rather than tag information for the `git-bug` repository.  I think I'd continue to ask the `Makefile` for this information.

Reply:
@smoyer64 is correct, we want to bake into the binary which code was compiled.

Reply:
@smoyer64 @MichaelMure  thank you for the detailed response, I see what you say.

I think I was too fast, opening this PR.

Here are my suggestions for now, would like to validate that with you, before I move forward:

1. Use `Makefile` in the first place (revert all the changes I did)
2. If `GitLastTag`, `GitCommit` and `GitExactTag` are not defined - try to extract info from `ReadBuildInfo` (if that's possible and Go is 1.18+)
3. If by the end we do not have required info - return some sort of better defined version, as now it's `-dev-`

There is one more thing wort mentioning: `ReadBuildInfo` does return only `vcs.revision`, that is commit, and we cannot get tag from there.

Does that sound okay for you?

Reply:
It seems that:
- `vcs.revision` is the equivalent of `GIT_COMMIT`
- `vcs.modified` tells if the original repo was dirty, which could be used to add a `-dirty` suffix on the version name

We are indeed missing the tag, which is quite annoying.

So ... what to do ... We could use those two variables when they are available, which I assume would at least partially fix #928. Ideally golang would also expose the tag, but I don't even see an issue in their tracker for that so it's not going to happen anytime soon.

Reply:
But yeah, the tag extraction in the makefile need to stay for now.

So we would have different version schema:
- `v0.8.0` <-- release
- `v0.8.0-dev-xyxyxy`  <-- non release, at commit xyxyxy
- `v0.8.0-dev-xyxyxy-dirty` <-- same but also dirty
- `dev-xyxyxy` <-- we don't know the tag
- `dev-xyxyxy-dirty` <-- same but also dirty

Reply:
There's also `vcs.time` ... after I wrote the above I was also coming to the conclusion that we might want a different format.  Fortunately, `debug.BuildInfo.Main.Version` will either contain the tag or the word "devel".  The real problem (for me) is that I'll never install a release package, and I'd love to know the "base" release for what I'm working on!

Reply:
Related (with links to a lot more):

- https://github.com/golang/go/issues/29228#issuecomment-457390055 (and whole issue)
- https://github.com/golang/go/issues/50603

Reply:
Note that you can still use linker flags with `go install` - would users do that if so instructed?

Reply:
Good reading about the build info and it's caveats.

Basically, the case when we can retrieve a tag from `debug.BuildInfo.Main.Version` will work only when we have a release, right?

> There's also vcs.time

We can use time from the build info as well, but will this more useful info for the use?
In this case, it sounds better to use linker flags as you suggest - it will be more useful for those who really need full release. But I should admit that will have to research that, since have never used linker flags

> In the examples above, the first three would have to be tagged AND would be produced by the Makefile right?

Yes, exactly. Updating PR description with that info again

For now, updated PR: reverted back changes in Makefile. The only change there, is `--dirty` flag. And added changes to return version per new schema described above